bin/pyang

```diff
@@ -413,6 +430,29 @@ Validates the YANG module in <filename> (or stdin), and all its dependencies."""
 
     ctx_validate_and_prune()
 
+    # verify the given features
+    for m in modules:
+        if m.arg in ctx.features:
+            if not o.exclude_features:
+                for f in ctx.features[m.arg]:
+                    if f not in m.i_features:
+                        sys.stderr.write("unknown feature %s in module %s\n" %
+                                         (f, m.arg))
+                        sys.exit(1)
+            # if --exclude-features, preserve the plugin interface by replacing
+            # '!feature' exclusions with 'feature' inclusions
+            else:
```

Comment:
This logic is wrong. I think it effectively just validates the features and strips the `!`, i.e., `ctx.features` will end up the same as in the `--features` case. This doesn't affect the operation of `--exclude-features`, it just affects what plugins will see in `ctx.features`. I hadn't added a test for that (but will).

Reply:
I believe that this has been fixed.